src/main/java/it/gov/pagopa/gpd/upload/repository/StatusRepository.java

```diff
@@ -64,11 +65,12 @@ public Status findStatusById(String id, String fiscalCode) {
             CosmosItemResponse<Status> response = container.readItem(id, new PartitionKey(fiscalCode), Status.class);
             return response.getItem();
         } catch (CosmosException ex) {
-            log.error("[Error][StatusRepository@findStatusById] The Status retrieval was not successful: " + ex.getStatusCode());
+            log.error("[Error][StatusRepository@findStatusById] The Status retrieval was not successful: {}", ex.getStatusCode());
             if(ex.getStatusCode() == HttpStatus.INTERNAL_SERVER_ERROR.getCode())
-                throw new AppException(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.name(), "Status saving unavailable");
-            else
-                throw new AppException(HttpStatus.valueOf(ex.getStatusCode()), String.valueOf(ex.getStatusCode()), "Status retrieval failed");
+                throw new AppException(HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.INTERNAL_SERVER_ERROR.name(), "The Status for given fileId " + id + " is not available");
+            else if(ex.getStatusCode() == NOT_FOUND.getCode())
+                throw new AppException(NOT_FOUND, "STATUS NOT FOUND", "The Status for given fileId " + id + " does not exist");
```

Comment:
Can we use format string like at 68?

Reply:
I will update it with `String.format(...)`